rpc/database.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Athanor Labs (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package rpc
+
+import (
+	"math/big"
+	"net/http"
+
+	"github.com/athanorlabs/atomic-swap/common/types"
+	mcrypto "github.com/athanorlabs/atomic-swap/crypto/monero"
+	"github.com/athanorlabs/atomic-swap/db"
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// RecoveryDB contains methods for retrieving swap recovery info from the database.
+type RecoveryDB interface {
+	GetContractSwapInfo(id types.Hash) (*db.EthereumSwapInfo, error)
+	GetSwapPrivateKey(id types.Hash) (*mcrypto.PrivateSpendKey, error)
+	GetCounterpartySwapPrivateKey(id types.Hash) (*mcrypto.PrivateSpendKey, error)
+}
+
+// DatabaseService ...
+type DatabaseService struct {
+	rdb RecoveryDB
+}
+
+// NewDatabaseService returns a new DatabaseService.
+func NewDatabaseService(rdb RecoveryDB) *DatabaseService {
+	return &DatabaseService{
+		rdb: rdb,
+	}
+}
+
+// GetContractSwapInfoRequest ...
+type GetContractSwapInfoRequest struct {
+	ID types.Hash `json:"id" validate:"required"`
+}
+
+// GetContractSwapInfoResponse ...
+type GetContractSwapInfoResponse struct {
+	StartNumber     *big.Int                   `json:"startNumber" validate:"required"`
+	SwapID          types.Hash                 `json:"swapID" validate:"required"`
+	Swap            *contracts.SwapCreatorSwap `json:"swap" validate:"required"`
+	SwapCreatorAddr ethcommon.Address          `json:"swapCreatorAddr" validate:"required"`
+}
+
+// GetContractSwapInfo returns the contract swap info for the given swap ID from the database.
+func (s *DatabaseService) GetContractSwapInfo(
+	_ *http.Request,
+	req *GetContractSwapInfoRequest,
+	resp *GetContractSwapInfoResponse,
+) error {
+	info, err := s.rdb.GetContractSwapInfo(req.ID)
+	if err != nil {
+		return err
+	}
+
+	resp.StartNumber = info.StartNumber
+	resp.SwapID = info.SwapID
+	resp.Swap = info.Swap
+	resp.SwapCreatorAddr = info.SwapCreatorAddr
+	return nil
+}
+
+// GetSwapPrivateKeyRequest ...
+type GetSwapPrivateKeyRequest struct {
+	ID types.Hash `json:"id" validate:"required"`
```

Comment:
Which of our many ID values is being passed here?

Reply:
`OfferID`/`SwapID`

Reply:
I don't understand. I just dug through the code and that ID is the OfferID. The SwapID is this: https://github.com/AthanorLabs/atomic-swap/blob/master/ethereum/utils.go#L58-L125.

Can we call the field `OfferID` (json `offerID`)?

Reply:
oh I've been referring to that ID as the contractSwapID lol. yeah I'll change it to offerID